presto-main/src/main/java/com/facebook/presto/execution/QueryManagerStats.java

```diff
@@ -167,15 +174,31 @@ public void stateChanged(QueryState newValue)
                     if (started) {
                         queryStopped();
                     }
-                    else {
+                    else if (queued) {
                         queryDequeued();
                     }
                     finalQueryInfoSupplier.get()
                             .ifPresent(QueryManagerStats.this::queryFinished);
                 }
+                else if (newValue.ordinal() == QUEUED.ordinal()) {
```

Comment:
I find the following easier to read:

```java
        @Override
        public void stateChanged(QueryState newValue)
        {
            synchronized (this) {
                if (stopped) {
                    return;
                }

                if (newValue.isDone()) {
                    stopped = true;
                    if (started) {
                        queryStopped();
                    }
                    else if (queued) {
                        queryDequeued();
                    }
                    finalQueryInfoSupplier.get()
                            .ifPresent(QueryManagerStats.this::queryFinished);
                    return;
                }

                if (newValue.ordinal() == QUEUED.ordinal()) {
                    if (!queued) {
                        queued = true;
                        queryQueued();
                    }
                }
                else if (newValue.ordinal() > QUEUED.ordinal()) {
                    if (queued) {
                        queryDequeued();
                        queued = false;
                    }
                    if (newValue.ordinal() >= RUNNING.ordinal()) {
                        if (!started) {
                            started = true;
                            queryStarted();
                        }
                    }
                }
            }
        }
    }
    ```

Reply:
@tdcmeehan Thanks for the review, I have made the change.